src/app/pages/courses/courses.component.ts

```diff
@@ -1,93 +1,94 @@
 import {
   Component,
   OnInit,
-  OnDestroy,
   ChangeDetectionStrategy,
-  ChangeDetectorRef,
-  NgZone
+  ChangeDetectorRef
 } from '@angular/core';
 import { Subscription } from 'rxjs';
 
 import { CourseItem } from '../../core/entities';
 import { CoursesService } from './courses.service';
 import { LoaderBlockService, ModalService } from '../../core/components';
-import { FilterPipe } from '../../core/pipes';
 
 @Component({
   selector: 'courses',
   templateUrl: 'courses.template.html',
   styleUrls: ['courses.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush,
-  providers: [
-    FilterPipe
-  ]
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
 
-export class CoursesComponent implements OnInit, OnDestroy {
+export class CoursesComponent implements OnInit {
   public coursesList: CourseItem[];
-  private originalCoursesList: CourseItem[];
-  private coursesSubscription: Subscription[];
+  public currentPage: number;
+  public itemsPerPage: number;
+  public coursesCount: number;
+  private searchPhrase: string;
 
   constructor(
       private coursesService: CoursesService,
       private cd: ChangeDetectorRef,
       private loaderBlockService: LoaderBlockService,
-      private ngZone: NgZone,
-      private modalService: ModalService,
-      private filterPipe: FilterPipe
+      private modalService: ModalService
   ) {
     this.coursesList = [];
-    this.originalCoursesList = [];
-    this.coursesSubscription = [];
+    
+    this.resetPagination();
```

Comment:
Is there any advantages to reset pagination in constructor over resetting in ngOnInit?